signing/logs/2023-06-06.md

```diff
@@ -0,0 +1,636 @@
+    }
+  },
+  {
+    "command": "rsa-generate",
+    "result": {
+      "success": true,
+      "id": "4E:16:DB:C8:67:4A:59:FA",
+      "label": "earlgrey_a0_prod_1"
+    }
+  },
+  {
+    "command": "rsa-generate",
+    "result": {
+      "success": true,
+      "id": "5D:61:F6:03:AE:33:6E:A2",
+      "label": "earlgrey_a0_prod_2"
+    }
+  },
+  {
+    "command": "rsa-generate",
+    "result": {
+      "success": true,
+      "id": "F5:DE:25:F4:78:7B:72:AE",
+      "label": "earlgrey_a0_dev_0"
+    }
+  },
+  {
+    "command": "rsa-generate",
+    "result": {
+      "success": true,
+      "id": "61:57:4F:AA:89:39:62:7F",
+      "label": "earlgrey_a0_dev_1"
+    }
+  },
+  {
+    "command": "rsa-generate",
+    "result": {
+      "success": true,
+      "id": "EB:32:96:00:FE:87:B1:B3",
+      "label": "earlgrey_a0_test_0"
+    }
+  },
+  {
+    "command": "rsa-generate",
+    "result": {
+      "success": true,
+      "id": "1B:A2:FF:CF:05:20:A7:15",
+      "label": "earlgrey_a0_test_1"
+    }
+  }
+]
+```
+
+### Export public keys
+
+The public keys were exported as DER files and then copied into `//sw/device/silicon_creator/rom/keys/real/rsa`.
+See the documentation in that directory for details about how the keys were converted into firmware resources.
+
+```bash
+(subshell)$ hsmtool --logging=info --token earlgrey_a0 -u user exec earlgrey_pubkey_export.hjson 
+[2023-06-06T23:23:07Z INFO  hsmtool::commands::exec] Executing command rsa-export
+[2023-06-06T23:23:07Z INFO  hsmtool::commands::exec] Executing command rsa-export
+[2023-06-06T23:23:07Z INFO  hsmtool::commands::exec] Executing command rsa-export
+[2023-06-06T23:23:07Z INFO  hsmtool::commands::exec] Executing command rsa-export
+[2023-06-06T23:23:07Z INFO  hsmtool::commands::exec] Executing command rsa-export
+[2023-06-06T23:23:07Z INFO  hsmtool::commands::exec] Executing command rsa-export
+[2023-06-06T23:23:07Z INFO  hsmtool::commands::exec] Executing command rsa-export
+[
+  {
+    "command": "rsa-export",
+    "result": {
+      "success": true
+    }
+  },
+  {
+    "command": "rsa-export",
+    "result": {
+      "success": true
+    }
+  },
+  {
+    "command": "rsa-export",
+    "result": {
+      "success": true
+    }
+  },
+  {
+    "command": "rsa-export",
+    "result": {
+      "success": true
+    }
+  },
+  {
+    "command": "rsa-export",
+    "result": {
+      "success": true
+    }
+  },
+  {
+    "command": "rsa-export",
+    "result": {
+      "success": true
+    }
+  },
+  {
+    "command": "rsa-export",
+    "result": {
+      "success": true
+    }
+  }
+]
+```
+
+### Generate Test Signatures
+
+The test signatures were copied into `//sw/device/silicon_creator/rom/keys/real/rsa`.
+See the documentation in that directory for details about how the test signatures are used in the unit tests.
+```bash
+(subshell)$ echo -n "test" >test.txt
+(subshell)$ hsmtool --logging=info --token earlgrey_a0 -u user exec earlgrey_test_signatures.hjson 
```

Comment:
Should all `hsmtool` config files be archived with the logs? Or alternatively, should `hstmtool` print the command in the result log?

Reply:
The config templates for keygen and export are already committed to the codebase under `//sw/device/silicon_creator/rom/keys/real/rsa/...`.  I'll add links from this document to those files so people reading the logs can easily find the corresponding inputs.